src/sql/platform/connection/common/connectionManagement.ts

```diff
@@ -80,7 +80,7 @@ export interface IConnectionManagementService {
 	/**
 	 * Opens the connection dialog to create new connection
 	 */
-	showConnectionDialog(params?: INewConnectionParams, model?: IConnectionProfile, connectionResult?: IConnectionResult): Promise<void>;
+	showConnectionDialog(params?: INewConnectionParams, options?: IConnectionCompletionOptions, model?: IConnectionProfile, connectionResult?: IConnectionResult): Promise<void>;
```

Comment:
The connection management service was ignoring the connection options when it hits the dialog which means it looses the settings defined by the caller.